lib/web_ui/lib/src/engine/browser_detection.dart

```diff
@@ -3,220 +3,51 @@
-
-/// The HTML engine used by the current browser.
-enum BrowserEngine {
-  /// The engine that powers Chrome, Samsung Internet Browser, UC Browser,
-  /// Microsoft Edge, Opera, and others.
-  ///
-  /// Blink is assumed in case when a more precise browser engine wasn't
-  /// detected.
-  blink,
-
-  /// The engine that powers Safari.
-  webkit,
-
-  /// The engine that powers Firefox.
-  firefox,
-}
-
-/// The signature of [getUserAgent].
-typedef UserAgentGetter = String Function();
-
-/// Returns the current user agent string.
-///
-/// This function is read-writable, so it can be overridden in tests.
-UserAgentGetter getUserAgent = defaultGetUserAgent;
-
-/// The default implementation of [getUserAgent].
-String defaultGetUserAgent() {
-  return domWindow.navigator.userAgent;
-}
-
-/// html webgl version qualifier constants.
-abstract class WebGLVersion {
-  /// WebGL 1.0 is based on OpenGL ES 2.0 / GLSL 1.00
-  static const int webgl1 = 1;
-
-  /// WebGL 2.0 is based on OpenGL ES 3.0 / GLSL 3.00
-  static const int webgl2 = 2;
-}
-
-/// Lazily initialized current browser engine.
-final BrowserEngine _browserEngine = _detectBrowserEngine();
-
-/// Override the value of [browserEngine].
-///
-/// Setting this to `null` lets [browserEngine] detect the browser that the
-/// app is running on.
-///
-/// This is intended to be used for testing and debugging only.
-BrowserEngine? debugBrowserEngineOverride;
-
-/// Returns the [BrowserEngine] used by the current browser.
-///
-/// This is used to implement browser-specific behavior.
-BrowserEngine get browserEngine {
-  return debugBrowserEngineOverride ?? _browserEngine;
-}
-
-BrowserEngine _detectBrowserEngine() {
-  final String vendor = domWindow.navigator.vendor;
-  final String agent = domWindow.navigator.userAgent.toLowerCase();
-  return detectBrowserEngineByVendorAgent(vendor, agent);
-}
-
-/// Detects browser engine for a given vendor and agent string.
-///
-/// Used for testing this library.
-@visibleForTesting
-BrowserEngine detectBrowserEngineByVendorAgent(String vendor, String agent) {
-  if (vendor == 'Google Inc.') {
-    return BrowserEngine.blink;
-  } else if (vendor == 'Apple Computer, Inc.') {
-    return BrowserEngine.webkit;
-  } else if (agent.contains('Edg/')) {
-    // Chromium based Microsoft Edge has `Edg` in the user-agent.
-    // https://docs.microsoft.com/en-us/microsoft-edge/web-platform/user-agent-string
-    return BrowserEngine.blink;
-  } else if (vendor == '' && agent.contains('firefox')) {
-    // An empty string means firefox:
-    // https://developer.mozilla.org/en-US/docs/Web/API/Navigator/vendor
-    return BrowserEngine.firefox;
-  }
-
-  // Assume Blink otherwise, but issue a warning.
-  print(
-      'WARNING: failed to detect current browser engine. Assuming this is a Chromium-compatible browser.');
-  return BrowserEngine.blink;
-}
+/// A flag to check if the current [operatingSystem] is a laptop/desktop
+/// operating system.
```

Comment:
Might wanna keep it consistent with `isMobile`?
```suggestion
/// A flag to check if the current browser is running on a laptop/desktop device.
```